codec-http2/src/main/java/io/netty/handler/codec/http2/WeightedFairQueueByteDistributor.java

```diff
@@ -111,6 +182,65 @@ public void updateStreamableBytes(StreamState state) {
     }
 
     @Override
+    public void updateDependencyTree(int childStreamId, int parentStreamId, short weight, boolean exclusive) {
+        if (weight < MIN_WEIGHT || weight > MAX_WEIGHT) {
+            throw new IllegalArgumentException(String.format(
+                    "Invalid weight: %d. Must be between %d and %d (inclusive).", weight, MIN_WEIGHT, MAX_WEIGHT));
+        }
+        if (childStreamId == parentStreamId) {
+            throw new IllegalArgumentException("A stream cannot depend on itself");
+        }
+
+        State state = state(childStreamId);
+        if (state == null) {
+            if (maxStateOnlySize == 0) {
```

Comment:
It is expected that the `maxStateOnlySize` may be exceeded in this method (see the `+2 because...` comment above the initialization of `stateOnlyRemovalQueue`). This check is to avoid using an empty queue and is currently OK because `maxStateOnlySize` is final. So this check says: "if we need to create a new state object that means we don't have a Http2Stream object and must retain the State in stateOnlyMap, but if stateOnlyMap is empty we should just drop the state"

Should I add a comment to this effect?